clang/test/SemaSYCL/intel-fpga-global-const.cpp

```diff
@@ -115,3 +115,11 @@
 //expected-note@+1{{conflicting attribute is here}}
 [[intel::bankwidth(8)]] extern const int var_bankwidth_2;
 [[intel::fpga_register]] const int var_bankwidth_2 =0;
+
+// Test that checks global constant variable (which allows the redeclaration) since
+// IntelFPGAConstVar is one of the subjects listed for [[intel::bank_bits()]] attribute.
+// Merging of different arg values.
+// expected-warning@+3{{attribute 'bank_bits' is already applied}}
+// expected-note@+1{{previous attribute is here}}
+[[intel::bank_bits(2, 3, 4, 5)]] extern const int bankbits_1;
+[[intel::bank_bits(41, 42, 43, 44)]] const int bankbits_1 = 0;
```

Comment:
I believe we will run into the same error in this case because we aren't checking argument values to determine whether this warning should be emitted or not. I think that is ok since otherwise we run into the issue of "what is same?" "Does the order matter?" etc.

Reply:
> I believe we will run into the same error in this case because we aren't checking argument values to determine whether this warning should be emitted or not. I think that is ok since otherwise we run into the issue of "what is same?" "Does the order matter?" etc.

/// Handle the bank_bits attribute.
// The values must be consecutive values (i.e. 3,4,5 or 2,1).

 Bank_bit attribute takes consecutive values, so order matters here.

Reply:
> I believe we will run into the same error in this case

I agree.   But I think that the current test case is deceptive in that it leads one to believe that it is a warning just because the values are different.   Hence the suggestion.